beacon-chain/p2p/gossipsub_peer_crawler.go

```diff
@@ -0,0 +1,515 @@
+package p2p
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/p2p/gossipsubcrawler"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/semaphore"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+)
+
+type peerNode struct {
+	id       enode.ID
+	isPinged bool
+	node     *enode.Node
+	peerID   peer.ID
+	topics   map[gossipsubcrawler.Topic]struct{}
+}
+
+type crawledPeers struct {
+	g *GossipsubPeerCrawler
+
+	mu       sync.RWMutex
+	byEnode  map[enode.ID]*peerNode
```

Comment:
`byEnode` and `byPeerId` have `*peerNode` as a key.
`byTopic` as a set of peer IDs as a key

==> `peerNodeByEnode`, `peerNodeByPid`, `pidsByTopic`?